syntax = "proto3";
package user;

service UserService {
  rpc GetUsers(GetUsersRequest) returns (GetUsersResponse) {}
  rpc GetUserById(GetUserByIdRequest) returns (GetUserByIdResponse) {}
  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {}
  rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse) {}
  rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse) {}
}

message User {
  int32 id = 1;
  string name = 2;
  string email = 3;
  string password = 4;
  string role = 5;
  string created_at = 6;
  string updated_at = 7;
}

message GetUsersRequest {
  
}

message GetUsersResponse {
  repeated User users = 1;
}

message GetUserByIdRequest {
  string id = 1;
}

message GetUserByIdResponse {
  User user = 1;
}

message CreateUserRequest {
  User user = 1;
}

message CreateUserResponse {
  User user = 1;
}

message UpdateUserRequest {
  User user = 1;
}

message UpdateUserResponse {
  User user = 1;
}

message DeleteUserRequest {
  string id = 1;
}

message DeleteUserResponse {
  User user = 1;
}   